can/interfaces/canalystii.py

```diff
@@ -63,17 +69,21 @@ def __init__(
 
         self.device = driver.CanalystDevice(device_index=device)
         for channel in self.channels:
-            if bit_timing:
-                try:
-                    if bit_timing.f_clock != 8_000_000:
-                        warnings.warn(
-                            f"bit_timing.f_clock value {bit_timing.f_clock} "
+            if isinstance(timing, BitTiming):
+                if timing.f_clock != 8_000_000:
+                    try:
```

Comment:
It's a trade-off in complexity between user code and library code. 

I like the idea of using a single timing instance across all interface implementations and with this code we come pretty close. From what i've seen, a BitTiming object with `f_clock`=8MHz could be compatible with (almost?) all interfaces just by changing the prescaler since most CAN 2.0 interfaces expect a multiple of 8MHz.

> someone somewhere is potentially going to be rather confused why their code is running with an incorrect clock value

I mean... maybe? In theory, if the f_clock conversion succeeds, it should have no negative effects. Should we throw a warning message if `f_clock` was changed?

Reply:
I think a warning sounds good. I created a utility method to unify the existing validation code in cantact and canalystii. It's pushed to the `bittiming` branch in my fork: https://github.com/lumagi/python-can/commit/dd505671a5fd9ca204733bf1b528c76739860def

Reply:
That looks good!

Reply:
I don't think I can add the commit to this PR. So feel free to cherry-pick it to the branch. Or should I open another PR for the commit?

Reply:
I cherry picked your commit and made some changes. Please take a look.